Add Option to Filter Events

angel23v opened the pull request:
## Quick Info

Improve two options to filter user events (by date or type), this in my events but if you go to Public Events that other users had created you also have a filter option but just by date because all are publics
There is a button to reset the info

## What does this change?
User Experience to get an event faster improving filters

## Why are you changing that?
Because its a better way to get an event, more comfortable and get a better user experience

## How do you manually test this?
Searching events using my filters options.  In my events and also in public events

## Screenshots 
# FIlter Options
![filters](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team4/assets/86389712/61b708fd-7022-438e-bab1-00ba5788dc5c)
# Result searched by public event
![result_Dos](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team4/assets/86389712/a3965b9f-2bed-4a31-84e0-12dbf11c607d)
# Result searched by date
![result](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team4/assets/86389712/6b5f5fd7-9601-4c57-b45e-5207143100d5)
# Public Events FIlter
![public_events_filter](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team4/assets/86389712/1230c980-6bbe-4bfb-9b4f-2d30df2d26a1)

angel23v committed:
Add Option to Filter Events

Improve two options to filter user events (by date or type), this in my events but if you go to Public Events that other users had created you also have a filter option but just by date, because all are publics

MauricioNC commented on app/views/layouts/_filter.html.erb:
```diff
@@ -0,0 +1,21 @@
+        <%= submit_tag "Filtrar Fecha", class: "filter" %>
+      <% end %>
+    </form>
+    <form>
+      <% if signed_in? && current_page?(events_index_path)%>
+        <%= select_tag(:privado, options_for_select([['Privado', true], ['Publico', false]]), class: 'check') %>
+        <%= submit_tag "Filtrar Tipo", class: "filter" %>
+      <% end %>
+
+    </form>
+      <% if signed_in? && current_page?(events_index_path)%>
+        <%= link_to 'Reiniciar', events_index_path, class: "filter" %>
+      <% end %>
```
You are evaluating same the conditions two times, you could do it in a single evaluation. You can improve the logic of this code too.

MauricioNC commented on app/views/events/public.html.erb:
```diff
@@ -3,13 +3,20 @@
+              <h5 class="card-title"> Dueño: <%= event.user.name %><%= image_tag "icons/user.svg" %></h5>
+              <h5 class="card-title"> Titulo: <%= event.titulo %><%= image_tag "icons/title.svg" %></h5>
+            </div>
+            <img>
+              <% if event.foto.attached? %>
+                <%= image_tag event.foto%>
+                <%= link_to 'Eliminar Foto', destroy_foto_event_path(event), method: :delete, data: { confirm: '¿Estás seguro de que quieres eliminar esta foto?' } %>
+              <% else %>
+                <%= 'No hay foto de este evento' %>
+              <% end %>
+            </img>
```
Same here.

MauricioNC replied:
And this the second cx

MauricioNC commented on app/views/events/index.html.erb:
```diff
@@ -1,19 +1,30 @@
           <% if event.foto.attached? %>
+            <%= image_tag event.foto%>
             <%= link_to 'Eliminar Foto', destroy_foto_event_path(event), method: :delete, data: { confirm: '¿Estás seguro de que quieres eliminar esta foto?' } %>
           <% else %>
             <%= 'No hay foto de este evento' %>
           <% end %>
-        <div class="card-body">
-          <h5 class="card-title"><%= event.id %>._ <%= event.titulo %></h5>
-          <p class="card-text"><%= event.descripcion %></p>
-        </div>
+        </img>
```
You could change the logic of this code, since you are wrapping an image tag inside another image tag if the condition is met.

MauricioNC replied:
Sorry, this requested change should be the first cx

MauricioNC commented on app/views/events/public.html.erb:
```diff
@@ -3,13 +3,20 @@
+            <img>
+              <% if event.foto.attached? %>
+                <%= image_tag event.foto%>
+                <%= link_to 'Eliminar Foto', destroy_foto_event_path(event), method: :delete, data: { confirm: '¿Estás seguro de que quieres eliminar esta foto?' } %>
```
Please use a single language, English or Spanish. (I mean the `destroy_foto_event_path`)

angel23v committed:
Refactor code on views

Change action, route & path to do it in a single language. Use of ternary operator to show image
